modules/aws/ecr.go

```diff
@@ -0,0 +1,93 @@
+package aws
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ecr"
+	"github.com/gruntwork-io/terratest/modules/testing"
+	"github.com/stretchr/testify/require"
+)
+
+// CreateECRRepo creates a new ECR Repository
+func CreateECRRepo(t testing.TestingT, region string, name string) *ecr.Repository {
+	repo, err := CreateECRRepoE(t, region, name)
+	require.NoError(t, err)
+	return repo
+}
+
+// CreateECRRepoE creates a new ECR Repository
+func CreateECRRepoE(t testing.TestingT, region string, name string) (*ecr.Repository, error) {
+	client := NewECRClient(t, region)
+	resp, err := client.CreateRepository(&ecr.CreateRepositoryInput{RepositoryName: aws.String(name)})
+	if err != nil {
+		return nil, err
+	}
+	return resp.Repository, nil
+}
+
+// GetECRRepo gets an ECR repository by name
+func GetECRRepo(t testing.TestingT, region string, name string) *ecr.Repository {
+	repo, err := GetECRRepoE(t, region, name)
+	require.NoError(t, err)
+	return repo
+}
+
+// GetECRRepoE gets an ECR Repository by name
+func GetECRRepoE(t testing.TestingT, region string, name string) (*ecr.Repository, error) {
+	client := NewECRClient(t, region)
+	repositoryNames := []*string{aws.String(name)}
+	resp, err := client.DescribeRepositories(&ecr.DescribeRepositoriesInput{RepositoryNames: repositoryNames})
+	if err != nil {
+		return nil, err
+	}
+	if len(resp.Repositories) != 1 {
+		return nil, fmt.Errorf("There is no repository named %s", name)
```

Comment:
NIT: We should return a custom error type here so that you can check for it. E.g., if one wanted to distinguish between error because repo doesn't exist and error because of AWS issues.

Reply:
This error condition should never occur. If no repositories are found, the `DescribeRepositories` call itself will return an error. We can't have more than one response because we're only providing one name to `RepositoryNames`. However, it's never a good idea to index a slice without length checking first, hence this check.. but I'm not sure how to cause this condition, so we also wouldn't be able to check for a custom error. Thus I just stuck with the `fmt.Errorf` (although I did adjust the wording). 

Note that this follows the convention already established throughout `terratest` (e.g. in ecs.go, ssm.go, and elsewhere).

Reply:
Ahhh ok. In that case, the error message should actually mention that this is an impossible case and if a user runs into it, they should file an issue.

Reply:
Good thinking, that sounds like the right thing to do. 2e7a21e